src/lib/url/url.hpp

```diff
@@ -92,6 +93,116 @@ class Url : public otUrl
      *
      */
     const char *GetProtocol(void) const { return mProtocol; }
+
+    /**
+     * Indicates whether or not the url contains the parameter.
+     *
+     * @param[in]  aName  A pointer to the parameter name.
+     *
+     * @retval TRUE   The url contains the parameter.
+     * @retval FALSE  The url doesn't support the parameter.
+     *
+     */
+    bool DoesParamExist(const char *aName) const { return (GetValue(aName) != nullptr); }
```

Comment:
How about `HasParam()`?

Reply:
I think the name `HasParam ()` is great. I have changed the name to `HasParam ()`.